### PR TITLE
Add touristic events participants

### PR DESCRIPTION
**Pull request by bouttier**

### CODE REVIEW

File: geotrek/tourism/migrations/0027_touristicevent_participants.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    dependencies = [
+        ('tourism', '0026_auto_20220907_1400'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='TouristicEventParticipantCategory',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('deleted', models.BooleanField(default=False, editable=False, verbose_name='Deleted')),
```
Comment (camillemonchicourt) [inline geotrek/tourism/migrations/0027_touristicevent_participants.py:18]: Actually, we don't use NoDeleteMixin for categories, and we plan to remove it where it is used (treks, signages...)

File: geotrek/tourism/models.py [hunk 2]
```diff
@@ -450,6 +451,10 @@ class TouristicEvent(ZoningPropertiesMixin, AddPropertyMixin, PublishableMixin,
     objects = TouristicEventManager()
     id_prefix = 'E'
 
+    @property
+    def participants_total(self):
+        return sum(map(attrgetter('count'), self.participants.all()))
```
Comment (Chatewgne) [inline geotrek/tourism/models.py:456]:
```suggestion
        return self.participants.annotate(participants_total(=Sum('count'))).participants_total
```

File: geotrek/tourism/forms.py [hunk 3]
```diff
@@ -101,3 +136,18 @@ def __init__(self, *args, **kwargs):
         # Since we use chosen() in trek_form.html, we don't need the default help text
         for f in ['themes', 'source']:
             self.fields[f].help_text = ''
+        participants_count = { p.category.pk: p.count for p in self.instance.participants.all() }
+        for category in TouristicEventParticipantCategory.objects.all():
```
Comment (Chatewgne) [inline geotrek/tourism/forms.py:140]:
```suggestion
        for category in TouristicEventParticipantCategory.objects.all().select_related('event', 'category'):
```

File: geotrek/tourism/forms.py [hunk 4]
```diff
@@ -101,3 +130,19 @@ def __init__(self, *args, **kwargs):
...
+            self.fieldslayout[0][1][1][0].append(field_id)
+
+    def _save_m2m(self):
+        super()._save_m2m()
+        for category in TouristicEventParticipantCategory.objects.all():
+            count = self.cleaned_data['participant_count_{}'.format(category.id)]
+            if count is not None:
+                TouristicEventParticipantCount.objects.update_or_create(event=self.instance, category=category, defaults={'count': count})
+            else:
+                TouristicEventParticipantCount.objects.filter(event=self.instance, category=category).delete()
```
Comment (Chatewgne) [inline geotrek/tourism/forms.py:148]: Please add tests for forms for these two scenarios
See other `test_forms.py` files

File: geotrek/tourism/tests/test_functional.py [hunk 5]
```diff
@@ -293,3 +299,53 @@ def test_custom_columns_mixin_on_export(self):
...
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(event.participants.count(), 2)
+
+        data.pop('participant_count_{}'.format(categories[0].pk))
+        response = self.client.post(event.get_update_url(), data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(event.participants.count(), 1)
+        count = event.participants.first()
+        self.assertEqual(count.count, 20)
+        self.assertEqual(count.category, categories[1])
```
Comment (Chatewgne) [inline geotrek/tourism/tests/test_functional.py:338]: Add label tests :

self.assertEqual(count.category.label, "Adultes")
self.assertEqual(count, "14 Adultes")

**Comment by bouttier** [comment]
There is still a duplicated scroll bar issue due to the tabbed layout of the update page. Curiously, this issue does not appear on trek update page although there is also a tabbed layout.

**Pull request merged**